### ISSUE TITLE
Local root dependency on batik causes classpath issue with deps.edn

### ISSUE THREAD

**Issue by respatialized** [quil/quil#389]
While running from the compiled JAR available via the Maven coordinates works, trying to use a version without a JAR distribution does not due to two :local/root dependencies on batik libraries in the project's [deps.edn](https://github.com/quil/quil/blob/919462a5474b8c0c4dbe3034b18349fe1f3b0771/deps.edn#L30-L31). I attempted to add exclusions for them, but that did not appear to resolve the classpath issue I encountered when trying to launch a REPL from within the project.

There may be another way to override these deps that I'm not familiar with (I'm not an expert with `deps.edn`), but I wanted to raise the issue here in case others encounter it.

## Example `deps.edn`:

```clojure
{:deps {quil/quil
        {:git/url "https://github.com/quil/quil" :sha "919462a5474b8c0c4dbe3034b18349fe1f3b0771"
           :exclusions [org.apache.xmlgraphics/batik-svggen org.apache.xmlgraphics/batik-dom]}
        ; {:mvn/version "4.0.0-SNAPSHOT"}
        org.apache.xmlgraphics/batik-svggen {:mvn/version "1.17"}
        org.apache.xmlgraphics/batik-dom {:mvn/version "1.17"}
        }}
```

**Comment by dgtized** [comment]
Currently quil does not support use as a deps.edn git/sha coordinate, it's only supported as a maven coordinate or built as a local jar. The batik dependency is one reason, but the primary cause is we bundle the core processing jars into the quil release so it works as a standalone release. Unfortunately, Processing *does not* release maven artifacts (see https://github.com/processing/processing/issues/5666), so it seems unlikely that the git/sha coordinate will work for quil in the foreseeable future.

However, I think should investigate using a maven coordinate for the batik dependencies as you have above. This might enable quil as a git/sha coordinate for ClojureScript targets using p5js, which do not depend on Processing at runtime.

Thanks for reporting this! This certainly exposes a few places we need clearer documentation.

**Comment by dgtized** [comment]
I took the liberty of inspecting your quil project, and I also noticed you were using the 4.0.0-SNAPSHOT release. We just released a new version to test which is the the 4.3-SNAPSHOT release which has updated Processing jars (which among other issues should work on M1 arch). That should contain everything in the latest sha minus a few test changes.

```
quil/quil {:mvn/version "4.3-SNAPSHOT"}
```

We intend to release that as a finished release soon provided we get more feedback and testing, and will update the docs to reflect that. Please test with the 4.3-SNAPSHOT over the 4.0.0-SNAPSHOT for now. Hopefully we can also add some automation to tag releases better so it won't feel necessary to use the sha release as a dependency.

Finally, it looks like you might be playing with rendering thi.ng/geom shapes with quil. I have a rough compatibility wrapper for that in https://github.com/dgtized/shimmers/blob/main/src/shimmers/common/quil_draws_geom.cljc, which works for a few cases, so might be useful. I'm a little focused on getting a stable release of Quil for now but I am very interested in improving compatibility between quil and the thi.ng libraries at a future date. It would likely be a compatibility library and not built into quil, but happy to hear ideas about that or to see what progress you make!

**Comment by jackrusher** [comment]
@respatialized I think @dgtized has already updated to the README in a PR, but setting up a `deps-new` template is very much on the TODO list! :)
